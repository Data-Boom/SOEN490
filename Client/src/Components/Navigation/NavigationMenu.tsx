/* eslint-disable react/display-name */

import { AppBar, Box, Divider, Drawer, Grid, IconButton, Toolbar, Typography, makeStyles } from "@material-ui/core"
import { ListRouter, getRoutedViews } from "./ListRouter"
import React, { useContext, useState } from 'react'

import { BrowserRouter } from 'react-router-dom'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'
import { Greeting } from "./Greeting"
import MenuIcon from '@material-ui/icons/Menu'
import { SessionTimeOut } from '../SessionTimeout'
import { UserContext } from "../../App"
import clsx from "clsx"
import { linkWidth } from './ListRouter'
import universitylogo from '../../Assets/universitylogo.png'

const drawerWidth = linkWidth

export default function NavigationMenu() {
  const { user } = useContext(UserContext)
  const [open, setOpen] = useState(false)
  const classes = useStyles()

  const handleDrawerOpen = () => {
    setOpen(true)
  }

  const handleDrawerClose = () => {
    setOpen(false)
  }

  const drawer = (): any => {
    return (
      <Drawer variant="persistent" anchor="left" open={open} className={classes.drawer} classes={{ paper: classes.drawerPaper }}>
        <div className={classes.drawerHeader}>
          <IconButton id='Close' onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </div>
        <Divider />
        {ListRouter()}
      </ Drawer>
    )
  }

  return (
    <>
      <BrowserRouter>
        <AppBar position="fixed" className={clsx(classes.appBar, { [classes.appBarShift]: open, })} color="primary">
          <Toolbar>
            <Grid container direction="row" justify="space-between" alignItems="center">
              <ClickAwayListener onClickAway={handleDrawerClose}>
                <Grid item>
                  <IconButton id='Open' edge="start" color="inherit" aria-label="menu" onClick={handleDrawerOpen} className={clsx(classes.menuButton)}>
                    <MenuIcon />
                  </IconButton>
                </Grid>
              </ClickAwayListener>
              <Grid item>
                <a href="https://concordia.ca"><img src={universitylogo} /></a>
              </Grid>
              <Grid container item xs={4} justify="flex-end">
                <Typography variant="h6" color="inherit">
                  Shock and Detonation Physics Database
                </Typography>
              </Grid>
              <Grid container item xs={4} justify="flex-end">
                <Greeting />
              </Grid>
            </Grid>
          </Toolbar>
          {user && user.sessionExpiration !== null &&
            <SessionTimeOut />
          }
        </AppBar>
        {drawer()}
        <Box className={clsx(classes.appBar, { [classes.appBarShift]: open, })} pt={16}>
          {getRoutedViews()}
        </Box>
      </BrowserRouter>
    </>
  )
}

//todo i dont like this useStyles() will refactor later, this is not a resuable component so whatever
const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    marginLeft: -drawerWidth,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  },
}))