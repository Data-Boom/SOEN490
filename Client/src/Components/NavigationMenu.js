/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */

import { AppBar, Box, Divider, Drawer, Grid, IconButton, List, ListItem, ListItemIcon, Toolbar, Typography, makeStyles } from "@material-ui/core"
import {
  HashRouter,
  NavLink,
  Route
} from "react-router-dom"
import { fileUploadRoute, graphRoute, homeRoute, searchRoute } from '../Consts/Routes'

import BarChartIcon from '@material-ui/icons/BarChart'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import CloudUploadIcon from '@material-ui/icons/CloudUpload'
import FileUploadView from "../Views/FileUploadView"
import GraphView from "../Views/GraphView"
import HomeIcon from '@material-ui/icons/Home'
import HomeView from '../Views/HomeView'
import MenuIcon from '@material-ui/icons/Menu'
import React from 'react'
import SearchIcon from '@material-ui/icons/Search';
import SearchView from "../Views/SearchView"
import clsx from "clsx"

const drawerWidth = 240

//todo i dont like this useStyles() will refactor later, this is not a resuable component so whatever
const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    marginLeft: -drawerWidth,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  },
}));


export default function NavigationMenu() {

  const [open, setOpen] = React.useState(false)
  const classes = useStyles()
  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  const renderNavLink = (route, title, icon) => {
    return (
      <ListItem button>
        <ListItemIcon>
          {icon}
        </ListItemIcon>
        <NavLink exact to={route}>
          {title}
        </NavLink>
      </ListItem>
    )
  }

  const drawer = () => {
    return (
      <Drawer variant="persistent" anchor="left" open={open} className={classes.drawer} classes={{
        paper: classes.drawerPaper,
      }}>
        <div className={classes.drawerHeader}>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </div>
        <Divider />
        <List>
          {renderNavLink(homeRoute, "Home", <HomeIcon />)}
          {renderNavLink(graphRoute, "Graph", <BarChartIcon />)}
          {renderNavLink(searchRoute, "Search", <SearchIcon />)}
          {renderNavLink(fileUploadRoute, "File Upload", <CloudUploadIcon />)}
        </List>
      </ Drawer>
    )
  }

  return (
    <>
      <HashRouter>
        <AppBar position="fixed" className={clsx(classes.appBar, { [classes.appBarShift]: open, })} color="primary">
          <Toolbar>
            <Grid container justify="space-between" alignItems="center">
              <Grid item>
                <IconButton edge="start" color="inherit" aria-label="menu" onClick={handleDrawerOpen} className={clsx(classes.menuButton)}>
                  {/* <IconButton edge="start" color="inherit" aria-label="menu" onClick={handleDrawerOpen} className={clsx(classes.menuButton, open && classes.hide)}> */}
                  <MenuIcon />
                </IconButton>
              </Grid>
              <Grid item>
                <Typography variant="h6" color="inherit">
                  Detonation Database
                </Typography>
              </Grid>
            </Grid>
          </Toolbar>
        </AppBar>
        {drawer()}
        <Box mt={16}>
          <Route exact path={homeRoute} component={HomeView} />
          <Route path={graphRoute} component={GraphView} />
          <Route path={fileUploadRoute} component={FileUploadView} />
        </Box>
        <Route path={searchRoute} component={SearchView} />
      </HashRouter >
    </>
  )
}