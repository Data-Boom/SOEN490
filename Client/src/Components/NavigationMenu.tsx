/* eslint-disable react/display-name */

import { AppBar, Box, Button, Divider, Drawer, Grid, IconButton, Toolbar, Typography, makeStyles } from "@material-ui/core"
import { HashRouter, Link } from 'react-router-dom'
import { ListRouter, getRoutedViews } from "./ListRouter"
import React, { useContext, useState } from 'react'

import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import MenuIcon from '@material-ui/icons/Menu'
import { UserContext } from "../App"
import clsx from "clsx"
import { linkWidth } from './ListRouter'
import { removeUserInStorage } from '../Common/Storage'
import { signInRoute } from "../Common/Consts/Routes"
import universitylogo from './universitylogo.png'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'


const drawerWidth = linkWidth

export default function NavigationMenu() {
  const { user, setUser } = useContext(UserContext)
  const [open, setOpen] = useState(false)
  const classes = useStyles()

  const handleDrawerOpen = () => {
    setOpen(true)
  }

  const handleDrawerClose = () => {
    setOpen(false)
  }

  function logout() {
    removeUserInStorage()
    window.location.replace("/")
  }

  const drawer = (): any => {
    return (
      <Drawer variant="persistent" anchor="left" open={open} className={classes.drawer} classes={{ paper: classes.drawerPaper }}>
        <div className={classes.drawerHeader}>
          <IconButton id='Close' onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </div>
        <Divider />
        {ListRouter()}
      </ Drawer>
    )
  }

  const renderGreeting = () => {
    return user && user.firstName ?
      (
        <Typography>
          Hello, {user.firstName} {user.lastName}
          <Button variant="contained" onClick={logout}>Sign out</Button>
        </Typography>
      ) : (
        <Button component={Link} to={signInRoute} id='btn1' variant="contained">Sign In</Button>
      )
  }

  return (
    <>
      <HashRouter>
        <AppBar position="fixed" className={clsx(classes.appBar, { [classes.appBarShift]: open, })} color="primary">
          <Toolbar>
            <Grid container direction="row" justify="space-between" alignItems="center">
              <ClickAwayListener onClickAway={handleDrawerClose}>
                <Grid item>
                  <IconButton id='Open' edge="start" color="inherit" aria-label="menu" onClick={handleDrawerOpen} className={clsx(classes.menuButton)}>
                    <MenuIcon />
                  </IconButton>
                </Grid>
              </ClickAwayListener>
              <Grid item>
                <img src={universitylogo} />
              </Grid>
              <Grid container item xs={4} justify="flex-end">
                <Typography variant="h6" color="inherit">
                  Shock and Detonation Physics Database
                </Typography>
              </Grid>
              <Grid container item xs={4} justify="flex-end">
                {renderGreeting()}
              </Grid>
            </Grid>
          </Toolbar>
        </AppBar>
        {drawer()}
        <Box className={clsx(classes.appBar, { [classes.appBarShift]: open, })} pt={16}>
          {getRoutedViews()}
        </Box>
      </HashRouter>

    </>
  )
}

//todo i dont like this useStyles() will refactor later, this is not a resuable component so whatever
const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    marginLeft: -drawerWidth,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  },
}))