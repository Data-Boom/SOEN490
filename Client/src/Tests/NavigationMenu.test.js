import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from 'enzyme';
import NavigationMenu from '../Components/NavigationMenu'

describe('NaigationMenu Component', () => {
    it("should match snapshot", () => {
        const tree = renderer.create(<NavigationMenu />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});