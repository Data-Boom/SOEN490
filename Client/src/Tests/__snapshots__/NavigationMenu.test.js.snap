// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NaigationMenu Component should match snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            aria-label="menu"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Detonation Database
          </h6>
        </div>
      </div>
    </div>
  </header>,
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-7 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <div
        className="makeStyles-drawerHeader-8"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <a
            aria-current="page"
            className="active"
            href="#/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 9.2h3V19H5zM10.6 5h2.8v14h-2.8zm5.6 8H19v6h-2.8z"
              />
            </svg>
          </div>
          <a
            aria-current={null}
            href="#/graph"
            onClick={[Function]}
          >
            Graph
          </a>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <a
            aria-current={null}
            href="#/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
              />
            </svg>
          </div>
          <a
            aria-current={null}
            href="#/uploadFile"
            onClick={[Function]}
          >
            File Upload
          </a>
        </div>
      </ul>
    </div>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-11"
  >
    <div
      className="Splash"
    >
      <img
        src="splashimage.png"
        width="100%"
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <div
        className="MuiBox-root MuiBox-root-13"
      >
        <h1
          className="Header"
        >
          Graphs
        </h1>
        <p
          className="Paragraph"
        >
          Interactive Gaphs! Add/Delete data sets, change axis, zoom in and out, enjoy seeing the database in a whole new way!
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-14"
      >
        <h1
          className="Header"
        >
          Downloads
        </h1>
        <p
          className="Paragraph"
        >
          Downloads have never been so easy. Select the datasets you want, choose the format you need, preview for perfection and DOWNLOAD WITH A CLICK.
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-15"
      >
        <h1
          className="Header2"
        >
          Personalized Accounts
        </h1>
        <p
          className="Paragraph"
        >
          Get ready to have your own account, upload your own data set, save your preferences and collaborate with others on theirs.
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-16"
      >
        <h1
          className="Header2"
        >
          Search and Discover
        </h1>
        <p
          className="Paragraph"
        >
          Need to find a dataset? Search with filters, by gas, DOI, author, category, fuel, oxidizer, dilutent, temperature, and more!
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-17"
      >
        <h1
          className="Header"
        >
          Data Cell Analysis
        </h1>
        <p
          className="Paragraph"
        >
          Welcome to the future! Bringing computer vision to data cells, upload an image of your experimental data cells and let magic do the rest! Data cell size in a minute!
        </p>
      </div>
    </div>
    <p>
      
    </p>
  </div>,
]
`;
